### PR TITLE
Первая лабораторная, вариант 1, Бендюжко Татьяна

### PR DESCRIPTION
**Pull request by Tatiana567**

### CODE REVIEW

File: clang/labs/lab1/bend_tatiana/PrintFunctionNames.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#include "clang/Frontend/FrontendPluginRegistry.h"
```
Comment (m-ly4) [inline clang/labs/lab1/bend_tatiana/PrintFunctionNames.cpp]: Please fix clang-format suggestions: https://github.com/NN-complr-tech/llvm-nnsu-2024/actions/runs/8308358647/job/22738518905?pr=16#step:4:10

File: clang/labs/lab1/bend_tatiana/PrintFunctionNames.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+namespace {
+
+    class MyASTConsumer : public ASTConsumer {
+        CompilerInstance& Instance;
+
+    public:
+        MyASTConsumer(CompilerInstance& Instance)
+            : Instance(Instance) {}
+
+        void HandleTranslationUnit(ASTContext& context) override {
```
Comment (m-ly4) [inline clang/labs/lab1/bend_tatiana/PrintFunctionNames.cpp]: Please fix clang-tidy warnings: https://github.com/NN-complr-tech/llvm-nnsu-2024/actions/runs/8308358647/job/22738518824?pr=16#step:6:485

File: clang/labs/lab1/bend_tatiana/PrintFunctionNames.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    class PrintFunctionNamesAction : public PluginASTAction {
+    protected:
+        std::unique_ptr<ASTConsumer> CreateASTConsumer(CompilerInstance& CI,
+            llvm::StringRef) override {
+            return std::make_unique<MyASTConsumer>(CI);
+        }
+
+        bool ParseArgs(const CompilerInstance& CI,
+            const std::vector<std::string>& args) override {
+            return true;
```
Comment (m-ly4) [inline clang/labs/lab1/bend_tatiana/PrintFunctionNames.cpp]: Please add an option `help` and a test for it.

**Review by m-ly4 — CHANGES_REQUESTED** [review]

File: clang/test/lab1/bend_tatiana/cats.cpp [hunk 4]
```diff
@@ -0,0 +1,29 @@
+// RUN: %clang_cc1 -load %llvmshlibdir/Myfirstplugin%pluginext -plugin classprinter %s | FileCheck %s
+
+// CHECK: Empty
+class Empty {};
+
+// CHECK: Cat
+struct Cat {
+    // CHECK-NEXT: |_ variable
+    int variable;
```
Comment (m-ly4) [inline clang/test/lab1/bend_tatiana/cats.cpp:9]: Please add a check for static fields.

**Review by aobolensk — APPROVED** [review]

**Review by m-ly4 — APPROVED** [review]

**Pull request merged**